### PR TITLE
Tweak solve_least_squares

### PR DESCRIPTION
**Pull request by sylee957**
#### References to other Issues or PRs

#15178

#### Brief description of what is fixed or changed

As well as `solve`, I think `solve_least_squares` can make use of more efficient routines than computing inverses, and I simply replaced the default routine with solve, to make use of any linear solver implemented in and also by the fact that it already contains the naive inverse computation method.

I also added every least squares methods available in sympy.

I think the there is no test case specifically for this function though, but the specific tests for each individual solvers.

#### Release Notes

- matrices
  - Added options for `solve_least_squares` to select more methods.

### CODE REVIEW

**Comment by sylee957** [comment]
I think `LDLsolve` doesn't use any specific algorithm for solving least square problems. Should it be removed from the selection?

**Comment by smichr** [comment]
> Should it be removed from the selection

I think what you have is ok -- this provides a single entry to all the solver routines

**Comment by sylee957** [comment]
@smichr 

I think it can be better to put the method in `solve` instead after merging this, because it would still be accessible.
If the methods are not kind of specialized one-step solving algorithms for specified domains of matrices it would not be a right place.

On the other hand, I think someone had put the `LDL` method on sparse matrix solver
https://github.com/sympy/sympy/blob/36374a3f7f1c6440511c1b649e6546c80c36bae7/sympy/matrices/sparse.py#L771-L826
Which is even not a valid matrix inversing method for dense matrices.

**Comment by sylee957** [comment]
Though adding `LDL` as some direct access here can still be future-proof way, and solve the inconsistency with sparse module before taking a look at solve.

I think it can be taken off from documentation instead, like a silent patch, and it can eventually stripped off from here, or not with ease.

File: sympy/matrices/matrices.py [hunk 1]
```diff
@@ -4130,9 +4130,28 @@ def QRsolve(self, b):
...
+            Vector representing the right hand side of the linear equation.
+
+        method : string or boolean, optional
+            If set to ``'CH'``, ``cholesky_solve`` routine will be used.
+
+            If set to ``'QR'``, ``QRsolve`` routine will be used.
+
+            If set to ``'PINV'``, ``pinv_solve`` routine will be used.
+
+            Other linear solving methods may be used, which can be inherited
```
Comment (smichr) [inline sympy/matrices/matrices.py]: This is a little unlcear to me. Maybe this edit?

For any other method, a naive approach will be taken using the conjugate of self; the passed method will be passed to `solve` method.

**Comment by smichr** [comment]
@asmeurer , one travis test didn't run. I manually restarted it. It passed. The sympy.sympy [test](https://dev.azure.com/SymPy/SymPy/_build/results?buildId=936), however, is still showing "Expected — Waiting for status to be reported" even though (if you navigate to that page) the tests there are showing as passed. The only way I know to work around this is to close and open the PR...which reruns everything again. Any other suggestsion?

**Comment by smichr** [comment]
@sylee957 , maybe Aaron will have a better suggestion but thanks for doing the restart. This is now in.

**Pull request merged**